Fix clean-aindent M-backspace in term-mode

wizmer opened the pull request:
See issue: https://github.com/pmarinov/clean-aindent-mode/issues/8

CeleritasCelery commented on clean-aindent-mode.el:
```diff
@@ -271,7 +271,9 @@ is indented less than the current one. Unindents current line to
   (interactive "p")
   (if (not (clean-aindent--inside-indentp))
-      (kill-word (- arg))  ;; Original "C-backspace" key function
+      (if (string= "term-mode" major-mode)
```
No need to do string conversion here. This could be written as `(eq 'term-mode major-mode)`

Commit:
clean-aindent--bsunindent is now bound on [remap backward-kill-word]

It was previously bound on (kbd "M-DEL") but some mode have a different usage
of this key binding. What we really want is that clean-aindent--bsunindent is
called whenever backward-kill-word was called, regardless of the binding.
